datanode/sqlstore/referral_sets.go

```diff
@@ -117,73 +122,55 @@ func (rs *ReferralSets) AddReferralSetStats(ctx context.Context, stats *entities
 	return err
 }
 
-func (rs *ReferralSets) GetReferralSetStats(ctx context.Context, setID entities.ReferralSetID, atEpoch *uint64, referee *entities.PartyID) (*entities.ReferralSetStats, error) {
+func (rs *ReferralSets) GetReferralSetStats(ctx context.Context, setID entities.ReferralSetID, atEpoch *uint64, referee *entities.PartyID, pagination entities.CursorPagination) ([]entities.FlattenReferralSetStats, entities.PageInfo, error) {
 	defer metrics.StartSQLQuery("ReferralSets", "GetReferralSetStats")()
 	var (
-		stats []*entities.ReferralSetStats
-		query string
-		args  []interface{}
+		query    string
+		args     []interface{}
+		pageInfo entities.PageInfo
+	)
+
+	query = fmt.Sprintf(
+		`SELECT at_epoch,
```

Comment:
This looks very much like the view we dropped in the migrations, why should we not keep the view? (referral_set_stats)

Reply:
Two reasons:
1. There is no need for the view if we can get all from one table.
2. The view was missing some data, and to accommodate the query I reworked, working out with a join was harder than having this one.